### ISSUE TITLE
Also change screensaver background?

### ISSUE THREAD

**Issue by rolfdeenen** [UtkarshGpta/bing-desktop-wallpaper-changer#9]
Hi there. I just read about your script and installed it on Fedora (24). Works great so far. Thank! I do like to keep my screensaver background the same as my desktop wallpaper. I tried to hack your script to make it also change that, but have failed so far. I know no Python. Could you help me? I don't think it is very difficult when you know a little Python (which I don't) . I think it should use something like:

SCHEMA = 'org.gnome.desktop.screensaver'
KEY = 'picture-uri'

Hope you can help. Pointers as what you think needs to be changed are also appreciated. I am more than willing to do it, but I have failed so far...

regards,
Rolf

**Comment by rolfdeenen** [comment]
Ah, never mind. Fixed it. Made a thinking error. I'll post my hack later, when I have more time.

**Comment by lmr** [comment]
Implemented in 673c2a5762192de0f41be97a141095e1cf54b654, this can be closed.

**Comment by rolfdeenen** [comment]
Ok, here's what I did. Note that I have no Python experience. I just guessed what I should do. And it seems to work :-)
First I created an additional class, just after the class BackgroundChanger. It is almost the same as the BackgroundChanger, but it referres to the screensaver wallpaper...

```
...gsettings.apply()

class ScreensaverChanger():
    SCHEMA = 'org.gnome.desktop.screensaver'
    KEY = 'picture-uri'

    def change_screensaver(self, filename):
        gsettings = Gio.Settings.new(self.SCHEMA)
        gsettings.set_string(self.KEY, "file://" + filename)
        gsettings.apply()
```

Then I called the def_screensaver in the main " block" and called it ss_changer:

```
          ...urllib.urlretrieve(image_url, image_path)
          bg_changer = BackgroundChanger()
          bg_changer.change_background(image_path)
          ss_changer = ScreensaverChanger()
          ss_changer.change_screensaver(image_path)
          summary...
```

Done. Note that when testing this will only work when the wallpaper hasn't been downloaded yet. If it is, the main routine is skipped and your screensaver wallpaper isn't updated.

**Comment by rolfdeenen** [comment]
Although perhaps not everybody wants to have both wallpapers changed...

**Comment by lmr** [comment]
@rolfdeenen the functionality has already been implemented, just update and enjoy. I just need the repository owner @UtkarshGpta to close this.

**Comment by rolfdeenen** [comment]
Closed with commit [673c2a5](https://github.com/UtkarshGpta/bing-desktop-wallpaper-changer/commit/673c2a5762192de0f41be97a141095e1cf54b654). Thanks!

**Comment by rolfdeenen** [comment]
Closed with commit [673c2a5762192de0f41be97a141095e1cf54b654](https://github.com/UtkarshGpta/bing-desktop-wallpaper-changer/commit/bdd9c84d962a0c0edd00f8494222205cd76acf51). Thanks!
